napari/layers/base/base.py

```diff
@@ -1295,6 +1291,39 @@ def _world_to_displayed_data_ray(
         vector_data_ndisplay /= np.linalg.norm(vector_data_ndisplay)
         return vector_data_ndisplay
 
+    def _world_to_data_dims_displayed(
+        self, dims_displayed: List[int], ndim_world: int
+    ) -> np.ndarray:
+        """Convert the indices of the displayed dims from world coordinates
+        to the data coordinates.
+
+        This accounts for differences in dimensionality between the world
+        and the data coordinates. For example a world dims order of
+        [2, 1, 0, 3] would be [0, 1] for a layer that only has two dimensions
+        or [1, 0, 2] for a layer with three as that corresponds to the
+        relative order of the last two and three dimensions respectively
+
+        Parameters
+        ----------
+        dims_displayed : List[int]
+            The world displayed dimensions.
+        ndim_world : int
+            The number of dimensions in the world coordinate system.
+
+        Returns
+        -------
+        dims_displayed_data : np.ndarray
+            The displayed dimensions in data coordinates.
+        """
+        offset = ndim_world - self.ndim
+        order = np.array(dims_displayed)
+        if offset <= 0:
+            dims_displayed_data = list(range(-offset)) + list(order - offset)
+        else:
+            dims_displayed_data = list(order[order >= offset] - offset)
```

Comment:
In the sig and docstring you say you return an ndarray,  but here it's a list. And I guess you can return from within the if/else directly ? I'm not sure if it's cleaner or not.

Reply:
Good catch. I've updated the docstring and now return from within the if/else.